packages/push/examples/smallest/main.rs

```diff
@@ -0,0 +1,215 @@
+// An output for this problem is an `i64`.
+struct Output(i64);
+
+// This is an implementation of the "smallest" problem from Tom Helmuth's
+// software synthesis benchmark suite (PSB1):
+//
+// T. Helmuth and L. Spector. General Program Synthesis Benchmark Suite. In
+// GECCO '15: Proceedings of the 17th annual conference on Genetic and
+// evolutionary computation. July 2015. ACM.
+//
+// Problem Source: C. Le Goues et al., "The ManyBugs and IntroClass
+// Benchmarks for Automated Repair of C Programs," in IEEE Transactions on
+// Software Engineering, vol. 41, no. 12, pp. 1236-1256, Dec. 1 2015.
+// doi: 10.1109/TSE.2015.2454513
+//
+// This problem is quite easy if you have a `Min` instruction, but can be
+// more a bit more difficult without that instruction.
+fn main() -> Result<()> {
+    let mut rng = thread_rng();
+
+    let args = Args::parse();
+
+    let training_cases = training_cases(&mut rng);
+
+    let scorer = FnScorer(
+        |genome: &Plushy| -> TestResults<test_results::Error<i128>> {
+            score_genome(genome, &training_cases)
+        },
+    );
+
+    let lexicase = Lexicase::new(training_cases.len());
+
+    let instruction_set = instructions();
+
+    let gene_generator =
+        GeneGenerator::with_uniform_close_probability(instruction_set.into_distribution()?);
+
+    let population = gene_generator
+        .to_collection_generator(args.max_initial_instructions)
+        .with_scorer(scorer)
+        .into_collection_generator(args.population_size)
+        .sample(&mut rng);
+
+    ensure!(population.is_empty().not());
+
+    let best = Best.select(&population, &mut rng)?;
+    println!("Best initial individual is {best}");
+
+    let umad = Umad::new(0.1, 0.1, &gene_generator);
+
+    let make_new_individual = Select::new(lexicase)
+        .then(GenomeExtractor)
+        .then(Mutate::new(umad))
+        .wrap::<GenomeScorer<_, _>>(scorer);
+
+    let mut generation = Generation::new(make_new_individual, population);
+
+    for generation_number in 0..args.num_generations {
+        match args.run_model {
+            RunModel::Serial => generation.serial_next()?,
+            RunModel::Parallel => generation.par_next()?,
+        }
+
+        let best = Best.select(generation.population(), &mut rng)?;
+        println!("Generation {generation_number:4} best is {best}");
+
+        if best.test_results.total_result.error == 0 {
+            println!("SUCCESS");
+            break;
+        }
+    }
+
+    Ok(())
+}
+
+fn score_genome(
+    genome: &Plushy,
+    training_cases: &Cases<Input, Output>,
+) -> TestResults<test_results::Error<i128>> {
+    // The penalty value to use when an evolved program doesn't have an expected
+    // "return" value on the appropriate stack at the end of its execution.
+    const PENALTY_VALUE: i128 = 1_000;
+    let program = Vec::<PushProgram>::from(genome.clone());
+    let errors: TestResults<test_results::Error<i128>> = training_cases
```

Comment:
just remove the let errors = here and directly return it ( i think this might be due to the switch to i128 i did - if it is sorry I missed this )